lib/Dialect/Region/Transforms/BufferizableOpInterfaceImpl.cpp

```diff
@@ -0,0 +1,178 @@
+           "only tensor types expected");
+    // Assume all operands are read.
+    return true;
+  }
+
+  bool bufferizesToMemoryWrite(
+      ::mlir::Operation *op, ::mlir::OpOperand &opOperand,
+      const ::mlir::bufferization::AnalysisState &state) const {
+    assert(::mlir::isa<::mlir::RankedTensorType>(opOperand.get().getType()) &&
+           "only tensor types expected");
+    // Assume all operands are written to.
+    return true;
+  }
+
+  ::mlir::bufferization::AliasingValueList
+  getAliasingValues(::mlir::Operation *op, ::mlir::OpOperand &opOperand,
+                    const ::mlir::bufferization::AnalysisState &state) const {
+    // Assume no aliasing.
+    return {};
+  }
+
+  ::mlir::LogicalResult
+  bufferize(::mlir::Operation *op, ::mlir::RewriterBase &rewriter,
+            const ::mlir::bufferization::BufferizationOptions &options) const {
+    auto envOp = ::mlir::cast<region::EnvironmentRegionOp>(op);
+    if (noTensorsIn(envOp.getArgs()) && noTensorsIn(envOp.getResults())) {
+      // Nothing to do.
+      return ::mlir::success();
+    }
+    // Convert op arguments to memrefs.
+    ::mlir::SmallVector<::mlir::Value> newArguments;
+    if (failed(convertToBuffers(envOp.getArgs(), newArguments, rewriter,
+                                options))) {
+      return ::mlir::failure();
+    }
+    // Infer result memref types by converting yield op operands to memrefs
+    ::mlir::SmallVector<::mlir::Value> newResults;
+    if (failed(convertToBuffers(envOp.getBody()->getTerminator()->getOperands(),
+                                newResults, rewriter, options))) {
+      return ::mlir::failure();
+    }
+    ::mlir::TypeRange resTypes(newResults);
+    // Create new op via generic constructor, op will have an empty region.
+    rewriter.setInsertionPoint(op);
+    ::mlir::OperationState state(op->getLoc(), op->getName(), newArguments,
+                                 resTypes, op->getAttrs());
+    state.addRegion();
+    ::mlir::Operation *newOp = ::mlir::Operation::create(state);
+    // Move block from old op into the new op.
+    newOp->getRegion(0).getBlocks().splice(newOp->getRegion(0).begin(),
+                                           op->getRegion(0).getBlocks());
+    rewriter.insert(newOp);
+    ::mlir::bufferization::replaceOpWithBufferizedValues(rewriter, op,
+                                                         newOp->getResults());
+
+    return ::mlir::success();
+  }
+};
+
+/// Bufferization of region.env_region_yield. Replaced with a new yield that
+/// operates on memrefs.
+struct EnvironmentRegionYieldOpInterface
+    : public ::mlir::bufferization::BufferizableOpInterface::ExternalModel<
+          EnvironmentRegionYieldOpInterface, region::EnvironmentRegionYieldOp> {
+  bool bufferizesToMemoryRead(
+      ::mlir::Operation *op, ::mlir::OpOperand &opOperand,
+      const ::mlir::bufferization::AnalysisState &state) const {
+    return true;
```

Comment:
what is it reading?

Reply:
This implementation follows ` scf.yield` op bufferization: read is true, write is false etc.